mlos_bench/mlos_bench/storage/sql/schema.py

```diff
@@ -46,6 +46,9 @@ class DbSchema:
     A class to define and create the DB schema.
     """
 
+    _LEN_ID = 255
```

Comment:
Would prefer we revert back to the original lengths for now and have an override specifically for when MySQL or MariaDB are specified as the engine in `__init__` so we don't mess up existing ones or unnecessarily limit the length on certain items.

Reply:
the original length *was* 255 - except for the `optimization_target`, which was 1024. I'd just set it to 512 and call it a day rather than creating exceptions for certain DBs

Reply:
Gave an alternative method suggestion here:
https://github.com/motus/MLOS/pull/11

Reply:
merged, thank you! I like that we have separate constants for parameters and metrics. I've added another one for status - we'll have another table with status soon, so it will come in handy.

P.S. We don't need to override the values for MySQL, though - as long as the IDs' lengths are 512 or less, we are fine.

Reply:
Do we want to keep that logic in there for now anyways.  Or maybe just a comment in the __init__ so that we know how to override it in the future?  I can imagine this being a thing we run into again with different storage backends.

Reply:
I'd much prefer to keep the logic DB-agnostic and have `_ID_LEN` at some reasonable value that can work with any DB engine. Overriding logic that you propose is a bit more flexible, but it would prevent the users from migrating their existing data from, e.g., SQLite to MySQL

Reply:
I suppose that's true, but we also haven't tested every DB that sqlalchemy supports.
If some new one came along and required even shorter column lengths, would you propose we alter existing schemas to accommodate that (#649)? That'd potentially lose data silently but be hard to special case which schema modifications we automate and which we don't.
Flexible length seems like the more sustainable future to me. It means we can start with a large column length. If it's too large for a new DB we encounter, it's immediately known and we can fix it with an override without affecting any existing installs. If someone is migrating from one backend to another that is an explicitly manual operation (we don't manage that for them automatically), hence they're on the hook for any data loss due to column length differences.

Reply:
We discussed offline and decided to move forward as is.
In case it's necessary, we can pull this commit up again in the future:
https://github.com/motus/MLOS/pull/11/commits/2589bf2c24e067a72b91ad9bdb73f5e360a7ad8c